File: camerax_android/android/src/main/kotlin/dev/hebei/camerax_android/legacy/ImageFormat.kt
package dev.hebei.camerax_android.legacy

val Int.imageFormatArgs
    get() = when (this) {
        android.graphics.ImageFormat.JPEG -> ImageFormat.JPEG
        android.graphics.ImageFormat.JPEG_R -> ImageFormat.JPEG_R
        android.graphics.ImageFormat.YUV_420_888 -> ImageFormat.YUV420_888
        android.graphics.PixelFormat.RGBA_8888 -> ImageFormat.RGBA8888
        else -> throw IllegalArgumentException()
    }

File: camerax_android/lib/src/legacy/core/image_format.dart
import 'package:camerax_android/src/legacy/camerax.g.dart' as $native;
import 'package:camerax_platform_interface/camerax_platform_interface.dart'
    as $interface;

extension ImageFormatArgs on $interface.ImageFormat {
  $native.ImageFormat get obj {
    switch (this) {
      case $interface.ImageFormat.jpeg:
        return $native.ImageFormat.jpeg;
      case $interface.ImageFormat.jpegR:
        return $native.ImageFormat.jpegR;
      case $interface.ImageFormat.yuv420_888:
        return $native.ImageFormat.yuv420_888;
      case $interface.ImageFormat.rgba8888:
        return $native.ImageFormat.rgba8888;
    }
  }
}

extension ImageFormatObj on $native.ImageFormat {
  $interface.ImageFormat get args {
    switch (this) {
      case $native.ImageFormat.jpeg:
        return $interface.ImageFormat.jpeg;
      case $native.ImageFormat.jpegR:
        return $interface.ImageFormat.jpegR;
      case $native.ImageFormat.yuv420_888:
        return $interface.ImageFormat.yuv420_888;
      case $native.ImageFormat.rgba8888:
        return $interface.ImageFormat.rgba8888;
    }
  }
}
